### ISSUE TITLE
configuration in child directory can corrupt configuration in parent directory

### ISSUE THREAD

**Issue by ccutrer** [rubocop/rubocop#10263]
The general problem here is that rubocop is using different configuration for a single file depending on how it's run, and that configuration is wrong. I haven't completely nailed down everything that's affected, but it's at least a fairly simple repro.

Given the following four files:

`.rubocop.yml`:
```yml
require:
  - rubocop-rails

AllCops:
  TargetRubyVersion: 2.7
  NewCops: enable

Rails:
  Severity: error
```

`subdir/.rubocop.yml`:
```yml
inherit_from:
  - ../.rubocop.yml

Rails:
  Enabled: false
```

`test.rb`:
```ruby
# frozen_string_literal: true

User.where('name = ?', 'Gabe')
```

`subdir/test2.rb` (I'm not sure if this file is necessary in order for rubocop to process the configuration file in its directory):
```ruby
# frozen_string_literal: true

puts 'hi'
```

## Expected behavior

I would expect that rubocop will find an offense of `Rails/WhereEquals` regardless of where it's run. I.e. `rubocop`, `rubocop -A`, `rubocop -- test.rb`, `rubocop -A -- test.rb` should find a single offense each time.

## Actual behavior

If I run `rubocop` or `rubocop -A`, no offenses are found in either file (WRONG!). But if I run `rubocop -- test.rb`, the offense is found (RIGHT!). My supposition is that there's a bug in the inheritance mechanism that because the entire Rails department is disabled in the sub-directory that state leaks to the parent directory's copy of the configuration. But if you're running _just_ `test.rb`, the sub-directory's config is never loaded, not having a chance to corrupt the parent. I have reason to believe it's not as simple as that, though. In my full app, the only cops that seems to "regress" with this sort of configuration were all cops that are marked as pending. So it might be that Enabled is getting reset incorrectly somewhere. I'm also not sure if being a separate gem (rubocop-rails) is important at all - that just happened to be the department that I encountered this with.

## RuboCop version

```
1.22.3 (using Parser 3.0.2.0, rubocop-ast 1.13.0, running on ruby 2.7.2 x86_64-darwin20)
  - rubocop-rails 2.12.4
```

**Comment by jonas054** [comment]
I've been able to reproduce the problem. It's not related to `rubocop-rails` specifically, but to pending cops in general. A PR is coming.
